completions/rrdtool

```diff
@@ -8,6 +8,17 @@ _rrdtool()
     if ((${#words[@]} == 2)); then
         COMPREPLY=($(compgen -W 'create update updatev graph dump restore
             last lastupdate first info fetch tune resize xport' -- "$cur"))
+    elif [[ ${words[1]} == "restore" ]]; then
+        case "$prev" in
+            restore | -*)
+                _filedir xml
+                ;;
+            *.xml)
+                ;;
+            *)
+                _filedir rrd
+                ;;
+        esac
```

Comment:
I'm not sure the last two cases do the right thing. Eyeballing the `rrdrestore` man page on Debian 10:
```
       rrdtool restore filename.xml filename.rrd [--range-check|-r] [--force-overwrite|-f]
```
So it appears to me that after `*.xml` we should do `_filedir rrd`, and the `*)` case should be removed. A more thorough fix would be to do `_filedir xml` only if no `*.xml` was in the cmdline already (i.e. not trigger it with prev=restore or an option) etc. I'm fine either way.

Reply:
I had gone off `rrdtool restore` which says

	rrdtool restore [--range-check|-r] [--force-overwrite|-f] filename.xml filename.rrd

In testing it out it seems that the positions of the -f and -r options do not matter and can come before, in between, or after the filename arguments. I will rework this.

Reply:
Thanks in advance. It seems to me that the most important bit is the first non-option argument to be a *.xml, and the second non-option one a *.rrd in case you don't want to introduce option completions. That should be fairly easily doable using `_args`. Doing this all the way on top of that with option completion including duplicate avoidance, taking care of short options as aliases (but not offering them as completions) is a bit tedious at the moment